log.go

```diff
@@ -54,6 +55,18 @@ type EventLogger interface {
 	EventBegin(ctx context.Context, event string, m ...Loggable) *EventInProgress
 
 	EventBeginInContext(ctx context.Context, event string, m ...Loggable) context.Context
+
+	Start(ctx context.Context, name string) context.Context
+	StartFromParentState(ctx context.Context, name string, parent []byte) context.Context
+
+	Finish(ctx context.Context)
```

Comment:
Need to make sure to harden this against calling `Finish` multiple times (complaining *loudly* when we do that). We'll almost certainly finish the wrong context once in a while.

Reply:
I agree, but am not sure what that would look like -  that is, there could be no `span` associated with the passed `ctx` - in which case we do nothing because `span == nil`. Else multiple calls to `Finish` will result in the "double-finished" `span` being reported to the tracing client multiple times (at least that's what I gathered from reading the jaeger [code](https://github.com/jaegertracing/jaeger-client-go/blob/master/span.go#L200) - I believe the result could be tracer dependent), what are your thoughts on how to guard against this?

There was some discussion on it here - https://github.com/opentracing/opentracing.io/issues/85 - but nothing definitive on how to handle it.